substrate/frame/utility/src/lib.rs

```diff
@@ -68,9 +68,24 @@ use sp_core::TypeId;
 use sp_io::hashing::blake2_256;
 use sp_runtime::traits::{BadOrigin, Dispatchable, TrailingZeroInput};
 pub use weights::WeightInfo;
-
 pub use pallet::*;
 
+pub trait BatchHook {
+	fn on_batch_start() -> sp_runtime::DispatchResult;
+	fn on_batch_end() -> sp_runtime::DispatchResult;
+
+}
+impl BatchHook for () {
```

Comment:
```suggestion
/// Hooks that will be called for `batch` calls.
#[impl_trait_for_tuples::impl_for_tuples(30)]
pub trait BatchHook {
	/// Will be called before a batch is executed.
	fn on_batch_start() -> sp_runtime::DispatchResult;
	/// Will be called after the batch was executed.
	///
	/// Depending on the exact batch call used, it may not be called when a batch item failed.
	fn on_batch_end() -> sp_runtime::DispatchResult;
}

impl BatchHook for () {
```